e2e/test/scenarios/permissions/sandboxing/sandboxing-via-ui.cy.spec.ts

```diff
@@ -0,0 +1,285 @@
+import { USER_GROUPS } from "e2e/support/cypress_data";
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+
+import {
+  type SandboxPolicy,
+  configureSandboxPolicy,
+  configureUser,
+  createTestCards,
+  signInAsSandboxedUser,
+  testCardsIncludeGizmosAndWidgets,
+  testCardsOnlyIncludeGizmos,
+  sandboxingUser as user,
+} from "./helpers/e2e-sandboxing-helpers";
+
+const { H } = cy;
+
+const preparePermissions = () => {
+  H.blockUserGroupPermissions(USER_GROUPS.ALL_USERS_GROUP);
+  H.blockUserGroupPermissions(USER_GROUPS.COLLECTION_GROUP);
+  H.blockUserGroupPermissions(USER_GROUPS.READONLY_GROUP);
+};
+
+describe(
+  "admin > permissions > sandboxing (tested via the UI)",
+  { tags: "@external" },
+  () => {
+    beforeEach(() => {
+      H.restore("postgres-12");
+      cy.signInAsAdmin();
+      H.setTokenFeatures("all");
+      preparePermissions();
+      cy.intercept("POST", "/api/card").as("saveQuestion");
+      cy.createUserFromRawData(user);
+    });
+
+    describe("we can apply a sandbox policy", () => {
+      it("to a table filtered using a question as a custom view", () => {
+        const policy: SandboxPolicy = {
+          filterTableBy: "custom_view",
+          customViewType: "question",
+        };
+        createTestCards(policy);
+        signInAsSandboxedUser();
+        testCardsIncludeGizmosAndWidgets(policy);
```

Comment:
on second thought maybe "testCardsShowGizmosAndWidgets" would be clearer

Reply:
or:

* not `createTestCards` but `createCardsWithGizmosAndWidgets`

* not `testCardsIncludeGizmosAndWidgets` but `cardsHaveRowsWithGizmosAndWidgets`